core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -222,19 +226,41 @@ public long getSessionId() {
   }
 
   @Override
-  public LockBlockResult lockBlock(final long blockId) throws IOException, AlluxioException {
+  public LockBlockResult lockBlock(final long blockId, final LockBlockOptions options)
+      throws IOException, AlluxioException {
     return retryRPC(
-        new RpcCallableThrowsAlluxioTException<LockBlockResult, BlockWorkerClientService
-            .Client>() {
+        new RpcCallableThrowsAlluxioTException<LockBlockResult, BlockWorkerClientService.Client>() {
           @Override
           public LockBlockResult call(BlockWorkerClientService.Client client)
               throws AlluxioTException, TException {
-            return ThriftUtils.fromThrift(client.lockBlock(blockId, getSessionId()));
+            return ThriftUtils
+                .fromThrift(client.lockBlock(blockId, getSessionId(), options.toThrift()));
           }
         });
   }
 
   @Override
+  public LockBlockResult lockUfsBlock(final long blockId, final LockBlockOptions options)
+      throws IOException, AlluxioException {
+    long retryInterval = Constants.SECOND_MS;
+    long timeout = System.currentTimeMillis() + Configuration
+        .getLong(PropertyKey.USER_UFS_BLOCK_OPEN_TIMEOUT_MS);
+    while (true) {
```

Comment:
Your suggestion sleeps twice (once in retry.attemptRetry(), once in that catch). That seems to be weird to me.